### DISCUSSION TITLE
[V5] Best way to check if Redis connection exists

### DISCUSSION THREAD

**Discussion by vkaelin** [adonisjs/core#1760] [Help]
Hi!
I'm trying to find a good way to tell the developer that the redis connection isn't working without having to wait for the long default 20 retries (``maxRetriesPerRequest`` key in the redis config).

Here is my current code in the ``AppProvider.ts``:

```ts
import { IocContract } from '@adonisjs/fold'
import { Exception } from '@poppinss/utils'

export default class AppProvider {
  constructor (protected $container: IocContract) {
  }

  public register () {
  }

  public async boot () {
    // Check if Redis connection works
    const Redis = (await import('@ioc:Adonis/Addons/Redis')).default
    const redisConfig = (await import('@ioc:Adonis/Core/Config')).default.get('redis')
    const connection = redisConfig.connections[redisConfig.connection]

    try {
      await Redis.info()
    } catch (e) {
      throw new Exception(
        `No Redis connection registered on host ${connection.host}:${connection.port}`,
        500,
        'E_NO_REDIS_CONNECTION',
      )
    }
  }

  public shutdown () {
  }

  public ready () {
  }
}
```

But it's still trying to make the request 20 times.
I saw that ioredis package had a `redis.status` key but I can't find it in the Adonis Redis package.
Thanks!

**Comment by thetutlage** [comment]
Simply set the `maxRetriesPerRequest` to `0` and hence it will not retry again

**Reply by vkaelin** [reply]
Thanks! Should I set ``maxRetriesPerRequest`` to ``0`` in development and let the default ``20`` in production. Sounds safer ?
